apps/metadata-editor/src/app/edit/edit-page.component.ts

```diff
@@ -132,25 +138,29 @@ export class EditPageComponent implements OnInit, OnDestroy {
 
     // if we're on the /create route, go to /edit/{uuid} on first change
     if (this.route.snapshot.routeConfig?.path.includes('create')) {
-      this.facade.draftSaveSuccess$.pipe(take(1)).subscribe(() => {
-        this.router.navigate(['edit', currentRecord.uniqueIdentifier], {
-          replaceUrl: true,
+      this.subscription.add(
+        this.facade.draftSaveSuccess$.pipe(take(1)).subscribe(() => {
+          this.router.navigate(['edit', currentRecord.uniqueIdentifier], {
+            replaceUrl: true,
+          })
         })
-      })
+      )
     }
 
     // if the record unique identifier changes, navigate to /edit/newUuid
-    this.facade.record$
-      .pipe(
-        filter(
-          (record) =>
-            record?.uniqueIdentifier !== currentRecord.uniqueIdentifier
-        ),
-        take(1)
-      )
-      .subscribe((savedRecord) => {
-        this.router.navigate(['edit', savedRecord.uniqueIdentifier])
-      })
+    this.subscription.add(
```

Comment:
This isn't needed because it's a short-live observable (there is a `take(1)` in the pipe); maybe getting rid of the subscription.add() and mentioning this fact in a comment will avoid confusion in the future?

Reply:
As discussed, I've kept this due to the error when editing a draft record.